presto-orc/src/main/java/com/facebook/presto/orc/DictionaryCompressionOptimizer.java

```diff
@@ -298,10 +352,21 @@ public int getDictionaryBytes()
         public double getDictionaryBytesPerFutureRow()
         {
             checkState(!directEncoded);
-            int dictionaryEntries = dictionaryColumn.getDictionaryEntries();
-            double dictionaryBytesPerEntry = 1.0 * dictionaryColumn.getDictionaryBytes() / dictionaryEntries;
-            double dictionaryBytesPerFutureValue = (1.0 * (dictionaryEntries - pastDictionaryEntries) / (dictionaryColumn.getValueCount() - pastValueCount)) * dictionaryBytesPerEntry;
-            return dictionaryBytesPerFutureValue * dictionaryColumn.getNonNullValueCount() / rowCount;
+
+            int currentDictionaryEntries = dictionaryColumn.getDictionaryEntries();
+            int currentValueCount = dictionaryColumn.getValueCount();
+
+            // average size of a dictionary entry
+            double dictionaryBytesPerEntry = 1.0 * dictionaryColumn.getDictionaryBytes() / currentDictionaryEntries;
+
+            // average number of entries added per non-value over the "past" period
+            double dictionaryEntriesPerFutureValue = 1.0 * (currentDictionaryEntries - pastDictionaryEntries) / (currentValueCount - pastValueCount);
+
+            // ration of non-null rows to all rows
+            double averageNonNullValuesPerRow = 1.0 * dictionaryColumn.getNonNullValueCount() / rowCount;
+
+            // Expected size comes down to: chance for a non-null row * chance for a unique dictionary value * average bytes per dictionary entry
+            return dictionaryBytesPerEntry * dictionaryEntriesPerFutureValue * averageNonNullValuesPerRow;
```

Comment:
dictionary bytes per entry: dict bytes / dict entry
dictionary entries per future value: dict entry / value
average non null values per row: non-null value / value

Multiplying the three gives: dict byte * non-null value / value^2.
If we don't multiply the first two, we get: dict byte / value.

I believe `averageNonNullValuesPerRow` should not be included.